rust/pact_matching_ffi/src/models/consumer.rs

```diff
@@ -1,15 +1,37 @@
 //! FFI wrapper code for pact_matching::models::Consumer
-use libc::c_char;
-
-pub use pact_matching::models::Consumer;
 
 use crate::util::*;
 use crate::{as_ref, ffi_fn};
+use libc::c_char;
+
+pub use pact_matching::models::Consumer;
 
 ffi_fn! {
     /// Get a copy of this consumer's name.
+    ///
     /// The copy must be deleted with `string_delete`.
     ///
+    /// # Usage
+    ///
+    /// ```c
+    /// // Assuming `file_name` and `json_str` are already defined.
+    ///
+    /// MessagePact *message_pact = message_pact_new_from_json(file_name, json_str);
+    /// if (message_pact == NULLPTR) {
+    ///     // handle error.
+    /// }
+    ///
+    /// Consumer *consumer = message_pact_get_consumer(message_pact);
+    /// if (consumer == NULLPTR) {
+    ///     // handle error.
+    /// }
+    ///
+    /// char *name = consumer_get_name(consumer);
+    /// printf("%s\n", name);
```

Comment:
This example code doesn't have error handling for the return value of `consumer_get_name`.
Since the example did already check that `consumer` is not NULL, it appears that an error could only occur if the string allocation fails or the Rust string has a NULL byte.